### DISCUSSION TITLE
Update Solid's testing docs

### DISCUSSION THREAD

**Discussion by daelmaak** [solidjs/solid#2094] [Ideas]
Hi there, I was following https://www.solidjs.com/guides/testing and I think it can be improved, mainly by promoting Vitest by default. Check out short summary of my journey in my article https://dev.to/daelmaak/why-you-should-not-use-jest-w-solid-3i00. I am aware of new docs being created, but so far, they are missing the testing guide completely.  If you agree, I'd be happy to help.

**Comment by atilafassina** [comment]
Thanks for the initiative, and I'd love to give it a go.
@LadyBluenotes, the testing guide is one that we ended up not porting. How about @daelmaak rewrites it with our help?

I also totally agree Vitest + Testing-Library would be the best combo. Perhaps the old guide predates Vitest and that's why it's not even mentioned.

---

@danieljcafonso is our in-house testing-lib expert, so perhaps he could offer some help too!

**Reply by atk** [reply]
I did not want to start on updating the testing docs until I have solved the issues with different versions of solid inside a testing run using vitest. When the first testing guide was written, we couldn't yet use vitest because the module loader had some serious issues that were only solved later. That being said, I appreciate others joining the efforts, because testing is a vast topic and I can't handle it all on my own.

**Reply by danieljcafonso** [reply]
@atk is there anything we can do to support you right now?

**Reply by atk** [reply]
I'm just done with my PR to vite-plugin-solid, so documentation would be the next thing on my list to tackle. Whoever wants to pick up the topic is welcome to connect with me by any means.

**Reply by atk** [reply]
Which branch should I base the rewrite on?

**Comment by danieljcafonso** [comment]
> Thanks for the initiative, and I'd love to give it a go. @LadyBluenotes, the testing guide is one that we ended up not porting. How about @daelmaak rewrites it with our help?
> 
> I also totally agree Vitest + Testing-Library would be the best combo. Perhaps the old guide predates Vitest and that's why it's not even mentioned.
> 
> @danieljcafonso is our in-house testing-lib expert, so perhaps he could offer some help too!

Count me in to help 👍

**Reply by atk** [reply]
I started the work on https://github.com/atk/solid-docs-next/blob/new-testing-guide/src/routes/guides/testing-your-app.mdx, it is still in progress. If you have any ideas about the structure, you can contact me on discord or on here.
